conda/common/path.py

```diff
@@ -85,14 +83,13 @@ def tokenized_startswith(test_iterable, startswith_iterable):
     return all(t == sw for t, sw in zip(test_iterable, startswith_iterable))
 
 
-def get_all_directories(files):
+def get_all_directories(files: Iterable[str]) -> list[tuple[str]]:
     return sorted(set(tuple(f.split('/')[:-1]) for f in files) - {()})
```

Comment:
I'm confused about how this works with Windows

Reply:
is [`os.path.normpath`](https://docs.python.org/3/library/os.path.html#os.path.normpath) called earlier?

Reply:
Could it be looking at tarfile paths

Reply:
That's a little weird, when following the code path it has a ton of indirection via `PathsData`, but I think no `os.path.normpath` use.

Reply:
In `conda.models.records`
```
class PathsData(Entity):
    # from info/paths.json
    paths_version = IntegerField()
    paths = ListField(PathData)
```